guides/source/active_record_callbacks.md

```diff
@@ -608,23 +933,36 @@ class PictureFile < ApplicationRecord
 end
 ```
 
-NOTE: The `:on` option specifies when a callback will be fired. If you don't supply the `:on` option the callback will fire for every action.
+NOTE: The `:on` option specifies when a callback will be fired. If you don't
+supply the `:on` option the callback will fire for every action.
+
+WARNING. When a transaction completes, the `after_commit` or `after_rollback`
+callbacks are called for all models created, updated, or destroyed within that
+transaction. However, if an exception is raised within one of these callbacks,
+the exception will bubble up and any remaining `after_commit` or
+`after_rollback` methods will _not_ be executed. As such, if your callback code
+could raise an exception, you'll need to rescue it and handle it within the
+callback in order to allow other callbacks to run. <br><br> `after_commit` makes
+very different guarantees than `after_save`, `after_update`, and
+`after_destroy`. For example, if an exception occurs in an `after_save` the
+transaction will be rolled back and the data will not be persisted. However,
+anything that happens `after_commit` can guarantee the transaction has already
+been completed and the data was persisted to the database. More on
+[transactional callbacks](#transaction-callbacks) below. <br><br> In the context
```

Comment:
This might need some tweaking, as it's moved here with this warning, we're already within the "transaction callbacks" section. Here's a suggestion:

> However, during `after_commit` the data was already persisted to the database, and thus any exception won't roll anything back anymore. 

I think we can also move the last paragraph together here:

>  However, during `after_commit` the data was already persisted to the database, and thus any exception won't roll anything back anymore. Also note that the code executed within `after_commit` or `after_rollback` callbacks is itself not enclosed within a transaction.

Feel free to tweak or rewrite it as necessary.